aws/cloud-init/master.script.tpl

```diff
@@ -317,10 +302,76 @@ EOF
     until [ "$${status}" != "PENDING" ]; do
       sleep 5
       status=$(aws route53 get-change --id "$${change_id}" --query 'Changinfo.Status' --output text)
-      echo "Route53 ALIAS status for ${cluster_name}: $${status}"
+      echo "Route53 ALIAS status for ${ops_advertise_addr}: $${status}"
     done
     echo "DNS record creation completed"
 
+    # Provisioning of a SAML identity provider.
+    SAML_NAME=`aws ssm get-parameter --name /telekube/${cluster_name}/saml/name --region $EC2_REGION --query 'Parameter.Value' --output text --with-decryption`
+    SAML_ADMIN_GROUP=`aws ssm get-parameter --name /telekube/${cluster_name}/saml/admin-group --region $EC2_REGION --query 'Parameter.Value' --output text --with-decryption`
+    SAML_ENTITY_DESCRIPTOR=`aws ssm get-parameter --name /telekube/${cluster_name}/saml/entity-descriptor --region $EC2_REGION --query 'Parameter.Value' --output text --with-decryption`
+    if [ ! -z "$${SAML_ENTITY_DESCRIPTOR}" ]; then
+      cat <<EOF > saml.yaml
+kind: saml
+version: v2
+metadata:
+  name: sso
+spec:
+  acs: https://${ops_advertise_addr}/portalapi/v1/saml/callback
+  attributes_to_roles:
+  - name: groups
+    value: $${SAML_ADMIN_GROUP}
+    roles:
+    - '@teleadmin'
+  display: $${SAML_NAME}
+  entity_descriptor: |
+EOF
+      # This needs to be properly indented, hence the sed command
+      printf "$${SAML_ENTITY_DESCRIPTOR}" | sed 's/^/\ \ \ \ /' >> saml.yaml
+      gravity resource create saml.yaml
+      rm saml.yaml
+
+      gravity resource create <<EOF
+kind: authgateway
+version: v1
+spec:
+  authentication:
+    type: saml
+    second_factor: "off"
```

Comment:
Just curious, why does SAML require turning off 2FA?

Reply:
I don't know if SAML requires turning this off - this is the resource spec you send to me in Slack (on Oct 15), so I didn't question it 🙂